### PR TITLE
Fix columns

### PR DESCRIPTION
**Pull request by dtassone**

### CODE REVIEW

**Review by oliviertassinari — COMMENTED** [review]
What was wrong? From a distant perspective, it looks like the logic in useColumns is correct but we had a regression in the state logic

**Comment by dtassone** [comment]
> What was wrong? From a distant perspective, it looks like the logic in useColumns is correct but we had a regression in the state logic, hence the current fix is only hiding it, without going after the root issue.

Well it makes sense to call it on resize as columns look at the size of the grid, and the container state is changed according to the total column size...
As I mention this area needs refactoring

**Comment by dtassone** [comment]
I'm merging this one as it's blocking all the other PRs, we can open an issue or investigate separately.

**Comment by oliviertassinari** [comment]
…

File: packages/grid/_modules_/grid/hooks/features/columns/useColumns.ts [hunk 1]
```diff
@@ -226,5 +232,13 @@ export function useColumns(columns: Columns, apiRef: ApiRef): InternalColumns {
 
   useApiMethod(apiRef, colApi, 'ColApi');
 
+  const onResize = React.useCallback(
+    (size) => {
+      resetColumns(size.width);
+    },
+    [resetColumns],
+  );
+  useApiEventHandler(apiRef, RESIZE, onResize);
```
Comment (oliviertassinari) [inline packages/grid/_modules_/grid/hooks/features/columns/useColumns.ts:241]: From what I understand `gridState.containerSizes?.viewportSize.width` already depends on the resize event, so this whole block is duplicated logic.

**Pull request merged**